script/long-range-delivery-drone.lua

```diff
@@ -1294,4 +1368,49 @@ lib.on_load = function()
   script_data = storage.long_range_delivery_drone or script_data
 end
 
+lib.on_configuration_changed = function(changed_data)
+  local mod_changes = changed_data.mod_changes
+  local old_version_string
+  if mod_changes and mod_changes["Long_Range_Delivery_Drones"] and mod_changes["Long_Range_Delivery_Drones"]["old_version"] then
+    old_version_string = mod_changes["Long_Range_Delivery_Drones"]["old_version"]
+  else
+    return
+  end
+  local version_strings = util.split(old_version_string, ".")
+  local old_version = {}
+  for i=1, #version_strings do
+    old_version[i] = tonumber(version_strings[i])
+  end
+
+  if old_version[1] < 2 then
```

Comment:
I'd rather just have a bool in storage and check that ratehr than this complicated checking of the version, something like
```
if not storage.regenerate_data_migration then
  storage.regenerate_data_migration = true
  -- All the rest underneath
end
```